### DISCUSSION TITLE
Is it possible to do a tree json

### DISCUSSION THREAD

**Discussion by weathon** [fastapi/fastapi#3296] [Questions]
For example, this is my database:

&#x200B;

|Name|Visit Time|
|:-|:-|
|Meredith|Jun 4|
|Meredith|Jun 6|
|Christian|Jun 10|
|Christain|Jun 11|

And I want it to return this kind of json

    {
    "meredith":["Jun 4","Jun 6"],
    "christian":["Jun 10","jun 11"]
    }

And in my usage, i might have more layers, like name-date-storeid-order

**Comment by lesnik512** [comment]
So, let's start with model for response. This would work for provided example:
```python
from typing import Dict, List
from pydantic import BaseModel

class Response(BaseModel):
    __root__: Dict[str, List[str]]

r = Response.parse_obj({
  "meredith":["Jun 4","Jun 6"],
  "christian":["Jun 10","jun 11"]
})
print(r.json())
# '{"meredith": ["Jun 4", "Jun 6"], "christian": ["Jun 10", "jun 11"]}'
```
About more layers: an example is needed

**Comment by alukach** [comment]
Not sure what datastore that you're using, but for Postgres the [`json_agg`](https://www.postgresql.org/docs/current/functions-aggregate.html) function will build a JSON string:

```sql
WITH data AS (
    SELECT
        name, date
    FROM (
        VALUES ('Meredith', 'Jun 4'),
            ('Meredith', 'Jun 6'),
            ('Christian', 'Jun 10'),
            ('Christian', 'Jun 11')) d (name, date))
SELECT
    name,
    json_agg(date) dates
FROM
    data
GROUP BY
    name;
```

This will give you data like:

```
   name    |        dates
-----------+----------------------
 Meredith  | ["Jun 4", "Jun 6"]
 Christian | ["Jun 10", "Jun 11"]
(2 rows)
```

With this, something like Pydantic's [JSON type](https://pydantic-docs.helpmanual.io/usage/types/#json-type) can be used to parse the JSON into an array:

```py
from typing import Dict, List
from pydantic import BaseModel, Json

class Response(BaseModel):
    __root__: Dict[str, Json[List[str]]]

@app.get('/', response_model=Response)
def user_dates():
    return db.query(...)
```
